stac_fastapi/pgstac/core.py

```diff
@@ -39,17 +38,100 @@
 class CoreCrudClient(AsyncBaseCoreClient):
     """Client for core endpoints defined by stac."""
 
-    async def all_collections(self, request: Request, **kwargs) -> Collections:
-        """Read all collections from the database."""
+    async def all_collections(  # noqa: C901
+        self,
+        request: Request,
+        # Extensions
+        bbox: Optional[BBox] = None,
+        datetime: Optional[DateTimeType] = None,
+        limit: Optional[int] = None,
```

Comment:
Thoughts from call: consider adding `id` to the endpoint, despite it being omitted from the [Collection Search Extension](https://github.com/stac-api-extensions/collection-search?tab=readme-ov-file#query-parameters-and-fields).

Reply:
Isn't that the same as just calling /collections/:id ?

Reply:
The use-case for including an `ids` parameter would be to limit the scope of a search in the context of scoped authentication for a STAC API, but we discussed some more and it probably makes more sense to use the `filter` extension for injecting scope limits in a search request.

Reply:
Yes, `ids` would be a better parameter name than `id` to evoke the ability to provide multiple as a filter. And for our specific needs today, we will make use of `filter` as @hrodmn said.  I believe it was @bitner who suggested adding the ability to filter by ID parameter, I'll let him weigh in on whether we should go without.

Reply:
I think it makes sense just to be parallel to the items spec (and yes, "ids" plural which is how it works in items)

Reply:
If you add it, I'd recommend to add a separate conformance class for it so that clients actually know whether it's supported or not.

PS: ids is not included in collection search as we just inherit from OGC API - Records, which doesn't have it. It's orthogonal to how ids is not part of OGC API - Features for items. ids is a STAC-specific thing.

Reply:
Unless someone else feels strongly, I think we can leave `ids` out since it is not in the extension spec and if a user knows the collection IDs they could just make several calls to `/collections/{id}`.